Impossible simulated events?

Apparently we can simulate events with non-positive S2 acceptances? This can cause builds to fail non-deterministically:

```
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-4.3.1, py-1.7.0, pluggy-0.8.0
rootdir: /home/travis/build/FlamTeam/flamedisx, inifile:
collected 98 items
tests/test_inference.py ............                                     [ 12%]
tests/test_likelihood.py F.............................                  [ 42%]
tests/test_mock_inference.py ...                                         [ 45%]
tests/test_source.py ................................................... [ 97%]
.                                                                        [ 98%]
tests/test_utils.py .                                                    [100%]
=================================== FAILURES ===================================
___________________________ test_wimp_SR1_source[ER] ___________________________
xes = <flamedisx.er_nr_base.ERSource object at 0x7f23de9236a0>
    def test_wimp_SR1_source(xes):
        # test KeyError 't' issue, because of add_extra_columns bug
        lf = fd.LogLikelihood(sources=dict(er=fd.SR1ERSource,
                                           wimp=fd.SR1WIMPSource),
                              free_rates=('er', 'wimp'))
    
        d = lf.simulate(er_rate_multiplier=1.0,
                        wimp_rate_multiplier=0.)
>       lf.set_data(d)
tests/test_likelihood.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
flamedisx/likelihood.py:182: in set_data
    source.set_data(deepcopy(data[dname]))
flamedisx/source.py:190: in set_data
    self._check_data()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <flamedisx.x1t_sr1.SR1WIMPSource object at 0x7f23d1ffb978>
    def _check_data(self):
        super()._check_data()
        if not self.check_acceptances:
            return
        for sn in signal_name.values():
            s_acc = self.gimme(sn + '_acceptance',
                               data_tensor=None, ptensor=None, numpy_out=True)
            if np.any(s_acc <= 0):
>               raise ValueError(f"Found event with non-positive {sn} "
                                 f"acceptance: did you apply and configure "
                                 "your cuts correctly?")
E               ValueError: Found event with non-positive s2 acceptance: did you apply and configure your cuts correctly?
flamedisx/er_nr_base.py:457: ValueError
```

Just saw this again (in https://travis-ci.org/github/FlamTeam/flamedisx/jobs/706823474, but I'll restart the job so that deletes the original log). Exerpt:

```
___________________________ test_wimp_SR1_source[ER] ___________________________
xes = <flamedisx.er_nr_base.ERSource object at 0x7f9b20712588>
    def test_wimp_SR1_source(xes):
        # test KeyError 't' issue, because of add_extra_columns bug
        lf = fd.LogLikelihood(sources=dict(er=fd.SR1ERSource,
                                           wimp=fd.SR1WIMPSource),
                              free_rates=('er', 'wimp'))
    
        d = lf.simulate(er_rate_multiplier=1.0,
                        wimp_rate_multiplier=0.)
>       lf.set_data(d)
tests/test_likelihood.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
flamedisx/likelihood.py:206: in set_data
    source.set_data(deepcopy(data[dname]))
flamedisx/source.py:190: in set_data
    self._check_data()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <flamedisx.x1t_sr1.SR1WIMPSource object at 0x7f9b20770ac8>
    def _check_data(self):
        super()._check_data()
        if not self.check_acceptances:
            return
        for sn in signal_name.values():
            s_acc = self.gimme(sn + '_acceptance',
                               data_tensor=None, ptensor=None, numpy_out=True)
            if np.any(s_acc <= 0):
>               raise ValueError(f"Found event with non-positive {sn} "
                                 f"acceptance: did you apply and configure "
                                 "your cuts correctly?")
E               ValueError: Found event with non-positive s2 acceptance: did you apply and configure your cuts correctly?
```

--- a/flamedisx/x1t_sr1.py
+++ b/flamedisx/x1t_sr1.py
@@ -72,6 +72,11 @@ class SR1Source:
         mean_eff= g1 / (1. + 0.219)
         return mean_eff * s1_relative_ly
 
+    @staticmethod
+    def s2_acceptance(s2):
+        return tf.where((s2 < 100) | (s2 > 6000),
+                        tf.zeros_like(s2, dtype=fd.float_type()),
+                        tf.ones_like(s2, dtype=fd.float_type()))
 
 
 # ER Source for SR1
@@ -105,11 +110,6 @@ class SR1ERSource(SR1Source,fd.ERSource):
         return tf.clip_by_value(q2 * (tf.constant(1.,dtype=fd.float_type()) - tf.exp(-nq / q3_nq)),
                                 tf.constant(1e-4,dtype=fd.float_type()),
                                 float('inf'))
-    @staticmethod
-    def s2_acceptance(s2):
-        return tf.where((s2 < 100) | (s2 > 6000),
-                        tf.zeros_like(s2, dtype=fd.float_type()),
-                        tf.ones_like(s2, dtype=fd.float_type()))
 
 @export
 class SR1NRSource(SR1Source, fd.NRSource):

